File: plugin/WiderScrollbar/src/main/java/WiderScrollbar.java
import android.content.Context;
import android.text.Editable;
import android.text.InputType;
import android.text.TextWatcher;
import android.view.View;
import com.aliucord.annotations.AliucordPlugin;
import com.aliucord.api.SettingsAPI;
import com.aliucord.entities.Plugin;
import com.aliucord.fragments.SettingsPage;
import com.aliucord.utils.DimenUtils;
import com.aliucord.views.TextInput;
import com.discord.databinding.WidgetChatListBinding;
import com.discord.widgets.chat.list.WidgetChatList;
import com.discord.widgets.chat.list.model.WidgetChatListModel;

import java.lang.reflect.Method;

@SuppressWarnings("MISSING_DEPENDENCY_SUPERCLASS")
@AliucordPlugin
public class WiderScrollbar extends Plugin {
    private Method getBindingMethod;

    public WiderScrollbar() {
        settingsTab = new SettingsTab(PluginSettings.class, settings);
    }

    public static class PluginSettings extends SettingsPage {
        private final SettingsAPI settings;

        public PluginSettings(SettingsAPI settings) {
            this.settings = settings;
        }

        @Override
        public void onViewBound(View view) {
            super.onViewBound(view);

            setActionBarTitle("Wider Scrollbar");

            TextInput textInput = new TextInput(requireContext());
            textInput.getEditText().setMaxLines(1);
            textInput.getEditText().setInputType(InputType.TYPE_CLASS_NUMBER);
            textInput.getEditText().setText(String.valueOf(settings.getInt("scrollbarWidth", 50)));
            textInput.getEditText().addTextChangedListener(new TextWatcher() {
                @Override
                public void beforeTextChanged(CharSequence s, int start, int count, int after) {}

                @Override
                public void onTextChanged(CharSequence s, int start, int before, int count) {}

                @Override
                public void afterTextChanged(Editable s) {
                    try {
                        int width = Integer.parseInt(s.toString());
                        if (width != 0) {
                            settings.setInt("scrollbarWidth", width);
                        }
                    } catch (Throwable ignored) {
                        settings.setInt("scrollbarWidth", 50);
                    }
                }
            });
            addView(textInput);
        }
    }

    @Override
    public void start(Context context) {
        try {
            getBindingMethod = WidgetChatList.class.getDeclaredMethod("getBinding");
            getBindingMethod.setAccessible(true);
        } catch (Exception e) {
            logger.error("Failed to get binding method", e);
            return;
        }

        patcher.after(WidgetChatList.class, "configureUI", (callFrame) -> {
            try {
                WidgetChatList widget = (WidgetChatList) callFrame.thisObject;
                WidgetChatListBinding binding = (WidgetChatListBinding) getBindingMethod.invoke(widget);
                binding.getRoot().setScrollBarSize(settings.getInt("scrollbarWidth", DimenUtils.dpToPx(50)));
            } catch (Exception e) {
                logger.error("Failed to set scrollbar width", e);
            }
        }, WidgetChatListModel.class);
    }

    @Override
    public void stop(Context context) {
        patcher.unpatchAll();
    }
}

File: plugin/WiderScrollbar/src/main/kotlin/WiderScrollbar.kt
import android.content.Context
import android.text.Editable
import android.text.InputType
import android.text.TextWatcher
import android.view.View
import com.aliucord.annotations.AliucordPlugin
import com.aliucord.api.SettingsAPI
import com.aliucord.entities.Plugin
import com.aliucord.fragments.SettingsPage
import com.aliucord.patcher.after
import com.aliucord.utils.DimenUtils.dp
import com.aliucord.views.TextInput
import com.discord.databinding.WidgetChatListBinding
import com.discord.widgets.chat.list.WidgetChatList
import com.discord.widgets.chat.list.model.WidgetChatListModel

@Suppress("MISSING_DEPENDENCY_SUPERCLASS")
@AliucordPlugin
class WiderScrollbar : Plugin() {
    private val getBindingMethod = WidgetChatList::class.java
        .getDeclaredMethod("getBinding")
        .apply { isAccessible = true }

    private fun WidgetChatList.getBinding() = getBindingMethod(this) as WidgetChatListBinding

    init {
        settingsTab = SettingsTab(PluginSettings::class.java).withArgs(settings)
    }

    class PluginSettings(private val settings: SettingsAPI) : SettingsPage() {
        override fun onViewBound(view: View) {
            super.onViewBound(view)

            setActionBarTitle("Wider Scrollbar")

            val textInput = TextInput(requireContext()) // .apply { hint = "Scrollbar width in DP" }
            textInput.editText.apply {
                maxLines = 1
                inputType = InputType.TYPE_CLASS_NUMBER
                setText(settings.getInt("scrollbarWidth", 50).toString())
                addTextChangedListener(object : TextWatcher {
                    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
                    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
                    override fun afterTextChanged(s: Editable) {
                        try {
                            val width = s.toString().toInt()
                            if (width != 0) settings.setInt("scrollbarWidth", width)
                        } catch (ignored: Throwable) {
                            settings.setInt("scrollbarWidth", 50)
                        }
                    }
                })
            }
            addView(textInput)
        }
    }

    override fun start(context: Context) {
        patcher.after<WidgetChatList>("configureUI", WidgetChatListModel::class.java) {
            try {
                getBinding().root.scrollBarSize = settings.getInt("scrollbarWidth", 50.dp)
            } catch (e: Throwable) {
                logger.error(e)
            }
        }
    }

    override fun stop(context: Context) = patcher.unpatchAll()
}